jcl/src/java.base/share/classes/java/lang/String.java

```diff
@@ -8315,5 +8315,48 @@ public String resolveConstantDesc(MethodHandles.Lookup lookup) {
 	public <R> R transform(Function<? super String, ? extends R> f) {
 		return f.apply(this);
 	}
+
+	/**
+	 * Adjusts the indentation of each line of this string based on the value of n, 
+	 * and normalizes line termination characters. 
+	 * 
+	 * @param n
+	 *			  the number of spaces to indent the string	 
+	 *
+	 * @return the string with indentation adjusted and line endings normalized
+	 * 
+	 * @since 12
+	 */
+	public String indent(int n) {
+		Stream<String> lines = this.lines();
+        Iterator<String> iter = lines.iterator();
+        StringBuilder builder = new StringBuilder();
+        
+        while (iter.hasNext()) {
+           String currentLine = iter.next();
+
+           if (n > 0) {
+               for (int i = 0; i < n; i++) {
+                   currentLine = " " + currentLine;
+               }
+           }
+           else {
+               int start = 0;
+               for (int i = 0; i < -n; i++) {
+                    if (currentLine.length() > i && Character.isWhitespace(currentLine.charAt(i))) {
+                        start++;
+                    }
+                    else {
+                        break;
+                    }
+               }
+               currentLine = currentLine.substring(start);
+           }
+
+           builder.append(currentLine + "\n");
```

Comment:
It looks like this would just add an extra newline character to the line. not sure thats correct but it should be fairly obvious when you get a change to test against the RI (RI = reference implementation = hotspot)